modules/internal/ChapelArray.chpl

```diff
@@ -2027,7 +2030,28 @@ module ChapelArray {
     proc tail(): this._value.eltType {
       return this[this.domain.high];
     }
-
+    pragma "no doc"
+    proc resizeAllocRange(r: range, d2: domain, factor=arrayAsVecGrowthFactor, direction=1, grow=1) {
```

Comment:
Maybe make 'direction' and 'grow' param arguments and mark this function 'inline' ?

OTOH there is so much more going on in the callers of this function that the above may easily not make a difference.